lib/modules/datasource/unity3d-packages/index.spec.ts

```diff
@@ -0,0 +1,197 @@
+          "1.14.2": "2025-03-27T10:54:45.412Z"
+        }
+      }
+    `;
+
+    httpMock
+      .scope('https://custom.registry.com')
+      .get('/com.unity.xr.openxr')
+      .reply(200, data);
+
+    const res = await getPkgReleases({
+      datasource: Unity3dPackagesDatasource.id,
+      packageName: 'com.unity.xr.openxr',
+      registryUrls: ['https://custom.registry.com'],
+    });
+
+    expect(res).toEqual({
+      homepage:
+        'https://docs.unity3d.com/Packages/com.unity.xr.openxr@1.14/manual/index.html',
+      registryUrl: 'https://custom.registry.com',
+      releases: [
+        {
+          changelogUrl: undefined,
+          isStable: true,
+          registryUrl: 'https://custom.registry.com',
+          releaseTimestamp: '2025-03-27T10:54:45.412Z',
+          version: '1.14.2',
+        },
+      ],
+    });
+  });
+
+  it(`package with changelog content and url`, async () => {
+    const data = `
+      {
+        "versions": {
+          "1.14.3": {
+            "_upm": {
+              "changelog": "### Fixed\\n\\n* Fixed Multiview Render Regions feature regression."
+            },
+            "documentationUrl": "https://docs.unity3d.com/Packages/com.unity.xr.openxr@1.14/manual/index.html",
+            "version": "1.14.3"
+          },
+          "1.12.0-exp.1": {
+            "_upm": {
+            },
+            "documentationUrl": "https://docs.unity3d.com/Packages/com.unity.xr.openxr@1.12/manual/index.html",
+            "version": "1.12.0-exp.1"
+          },
+          "1.0.0-pre.1": {
+            "version": "1.0.0-pre.1"
+          },
+          "0.1.2-preview.2": {
+            "version": "0.1.2-preview.2"
+          }
+        },
+        "time": {
+          "1.14.3": "2025-04-18T18:06:12.036Z",
+          "1.12.0-exp.1": "2024-07-03T15:24:28.000Z",
+          "1.0.0-pre.1": "2021-02-11T19:26:19.000Z",
+          "0.1.2-preview.2": "2021-01-05T17:57:41.000Z"
+        }
+      }
+    `;
+
+    httpMock
+      .scope('https://packages.unity.com')
+      .get('/com.unity.xr.openxr')
+      .reply(200, data);
+
+    const res = await getPkgReleases({
+      datasource: Unity3dPackagesDatasource.id,
+      packageName: 'com.unity.xr.openxr',
+      registryUrls: ['https://packages.unity.com'],
+    });
+
+    expect(res?.homepage).toBe(
+      'https://docs.unity3d.com/Packages/com.unity.xr.openxr@1.14/manual/index.html',
+    );
+    expect(res?.registryUrl).toBe('https://packages.unity.com');
+
+    expect(res?.releases).toHaveLength(4);
+
+    expect(res?.releases[3]).toEqual({
+      changelogContent:
+        '### Fixed\n\n* Fixed Multiview Render Regions feature regression.',
+      changelogUrl:
+        'https://docs.unity3d.com/Packages/com.unity.xr.openxr@1.14/changelog/CHANGELOG.html',
+      isStable: true,
+      registryUrl: 'https://packages.unity.com',
+      releaseTimestamp: '2025-04-18T18:06:12.036Z',
+      version: '1.14.3',
+    });
+
+    expect(res?.releases[2]).toEqual({
```

Comment:
Why comparing each release separately? Can be done in a single object same for the other fields ie. homepage, regsitryUrl